packages/mermaid/src/diagrams/packet/renderer.ts

```diff
@@ -0,0 +1,94 @@
+import type { Diagram } from '../../Diagram.js';
+import type { PacketDiagramConfig } from '../../config.type.js';
+import type { DiagramRenderer, DrawDefinition, Group, SVG } from '../../diagram-api/types.js';
+import { selectSvgElement } from '../../rendering-util/selectSvgElement.js';
+import type { PacketDB, PacketWord } from './types.js';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const draw: DrawDefinition = (_text, id, _version, diagram: Diagram) => {
+  const db = diagram.db as PacketDB;
+  const config = db.getConfig();
+  const { rowHeight, paddingY, bitWidth, bitsPerRow } = config;
+  const words = db.getPacket();
+  const title = db.getDiagramTitle();
+  const totalRowHeight = rowHeight + paddingY;
+  const svgHeight = totalRowHeight * (words.length + 1) - (title ? 0 : rowHeight);
+  const svgWidth = bitWidth * bitsPerRow + 2;
+  const svg: SVG = selectSvgElement(id);
+
+  svg.attr('width', '100%');
+  svg.attr('viewbox', `0 0 ${svgWidth} ${svgHeight}`);
```

Comment:
Maybe do a similar approach to `pie`?

<https://github.com/mermaid-js/mermaid/blob/e8ee5f548fdda51f246802473992e479c4cd82ee/packages/mermaid/src/diagrams/pie/pieRenderer.ts#L52-L53>

Something like this?

```suggestion
  configureSvgSize(svg, svgHeight, svgWidth, config.useMaxWidth);
  svg.attr('viewbox', `0 0 ${svgWidth} ${svgHeight}`);
```

Reply:
This was why I added `width: 100%` directly. 
https://github.com/mermaid-js/mermaid/issues/5038

But for now, I'll use the existing convention, and we can fix everything together as part of #5038.